src/routes/oauth/config.ts

```diff
@@ -0,0 +1,387 @@
+        jwt.sign(
+          {
+            iss: process.env.AUTH_PROVIDER_APPLE_TEAM_ID,
+            aud: 'https://appleid.apple.com',
+            sub: process.env.AUTH_PROVIDER_APPLE_CLIENT_ID,
+          },
+          process.env.AUTH_PROVIDER_APPLE_PRIVATE_KEY.replace(/\\n/g, '\n'),
+          {
+            algorithm: 'ES256',
+            header: {
+              kid: process.env.AUTH_PROVIDER_APPLE_KEY_ID,
+              typ: undefined,
+              alg: 'ES256',
+            },
+            expiresIn: '180d',
+          }
+        ),
+      scope: ['name', 'email'],
+      custom_params: {
+        response_type: 'code id_token',
+        response_mode: 'form_post',
+      },
+    },
+    profile: ({ jwt }) => {
+      const payload = jwt?.id_token?.payload;
+      // * See https://developer.apple.com/forums/thread/118209
+      const displayName = payload?.name
+        ? `${payload.name.firstName} ${payload.name.lastName}`
+        : payload.email;
+      return {
+        id: payload.sub,
+        displayName,
+        email: payload.email,
+        emailVerified: payload.email_verified === 'true',
+      };
+    },
+  },
+
+  azuread: {
+    grant: {
+      oauth: 2,
+      scope_delimiter: ' ',
+      client_id: process.env.AUTH_PROVIDER_AZUREAD_CLIENT_ID,
+      client_secret: process.env.AUTH_PROVIDER_AZUREAD_CLIENT_SECRET,
+      authorize_url: `${azureBaseUrl}/[subdomain]/oauth2/authorize`,
+      access_url: `${azureBaseUrl}/[subdomain]/oauth2/token`,
+      profile_url: `${azureBaseUrl}/[subdomain]/openid/userinfo`,
+      subdomain: process.env.AUTH_PROVIDER_AZUREAD_TENANT || 'common',
+    },
+    profile: ({ jwt }) => {
+      const payload = jwt?.id_token?.payload;
+      return {
+        id: payload.oid,
+        displayName: payload.name,
+        email: payload.email,
+      };
+    },
+  },
+
+  bitbucket: {
+    grant: {
+      client_id: process.env.AUTH_PROVIDER_BITBUCKET_CLIENT_ID,
+      client_secret: process.env.AUTH_PROVIDER_BITBUCKET_CLIENT_SECRET,
+      scope: ['account'],
+    },
+    profile: async ({ profile, access_token }) => {
+      const {
+        data: {
+          values: [{ email, is_confirmed }],
+        },
+      } = await axios.get<{
+        values: { email: string; is_confirmed: boolean }[];
+      }>(`https://api.bitbucket.org/2.0/user/emails`, {
+        headers: { Authorization: `Bearer ${access_token}` },
+      });
+      return {
+        id: profile.uuid,
+        displayName: profile.display_name,
+        emailVerified: is_confirmed,
+        email,
+        avatarUrl: profile.links.avatar.href,
+      };
+    },
+  },
+
+  discord: {
+    grant: {
+      client_id: process.env.AUTH_PROVIDER_DISCORD_CLIENT_ID,
+      client_secret: process.env.AUTH_PROVIDER_DISCORD_CLIENT_SECRET,
+      scope: ['identify', 'email'],
+    },
+    profile: ({ profile }) => ({
+      id: profile.id,
+      displayName: `${profile.username}#${profile.discriminator}`,
```

Comment:
I wonder if we should skip `#${profile.discriminator}` here.

If I sign in to an app using Discord I think I would prefer my display name on that app to be `elitan` instead of `elitan#7162`.

Reply:
0 opinion :)
Maybe someone else? @nunopato ?

Reply:
Although I have a tiny opinion: this information will be a bit complicated for developers to get at a later stage, while it will be easier for them either to modify the display name in the database or to hide the discriminator in their frontend

Reply:
Ok, let's keep it as is for now.

Reply:
Seems like we had this behaviour before too: https://github.com/oauth-everything/passport-discord/blob/master/src/Strategy.ts#L91

Let's keep it as is.